### PR TITLE
Add Cilium proxy visibility annotations to pods

### PR DESCRIPTION
**Pull request by g-linville**
These are needed in order to get HTTP metrics from Cilium.

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/runtime/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/runtime/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/controller/appdefinition/deploy.go [hunk 1]
```diff
@@ -522,6 +524,18 @@ func addPrometheusAnnotations(annotations map[string]string, container v1.Contai
...
 
+func addCiliumAnnotations(annotations map[string]string, container v1.Container) {
+	var visibilityConfigs []string
+	for _, port := range container.Ports {
+		if port.Protocol == v1.ProtocolHTTP {
+			visibilityConfigs = append(visibilityConfigs, fmt.Sprintf("<Ingress/%d/TCP/HTTP>", port.TargetPort))
+		}
+	}
+	if len(visibilityConfigs) > 0 {
+		annotations[labels.CiliumProxyVisibility] = strings.Join(visibilityConfigs, ",")
```
Comment (thedadams) [inline pkg/controller/appdefinition/deploy.go:535]: What are the chance that this exceeds 253 characters?

**Review by thedadams — APPROVED** [review]

**Review by iwilltry42 — APPROVED** [review]

File: pkg/labels/labels.go [hunk 2]
```diff
@@ -71,6 +71,8 @@ const (
 	PrometheusScrape = "prometheus.io/scrape"
 	PrometheusPath   = "prometheus.io/path"
 	PrometheusPort   = "prometheus.io/port"
+
+	CiliumProxyVisibility = "policy.cilium.io/proxy-visibility"
```
Comment (tylerslaton) [inline pkg/labels/labels.go:75]: Think its worth using this package?

https://github.com/cilium/cilium/blob/696167cb625e3b61b42412a37f7b02adcce2ff3f/pkg/annotation/k8s.go#L107

**Review by tylerslaton — APPROVED** [review]

File: pkg/controller/appdefinition/deploy.go (hunk 1 above)
Comment (iwilltry42) [reply pkg/controller/appdefinition/deploy.go:535]: @thedadams Isn't 253 the character limit of the annotation prefix only?
I'm not aware on a limit for the annotation value :thinking: 
I just know that the total size of all keys and values must not exceed  256 KB.

Comment (g-linville) [reply pkg/controller/appdefinition/deploy.go:535]: I just tested this with like 15 ports defined on one container (each is five digits long) and the annotation is created properly with no issues.

File: pkg/labels/labels.go (hunk 2 above)
Comment (g-linville) [reply pkg/labels/labels.go:75]: I just tried it out but there are multiple mutual dependencies between us and Cilium, so adding this updated some of those and I'm a little afraid of causing a problem and would just prefer to have this in our code.

**Pull request merged**